libfreerdp/core/capabilities.c

```diff
@@ -2137,11 +2139,30 @@ static BOOL rdp_apply_virtual_channel_capability_set(rdpSettings* settings, cons
 	WINPR_ASSERT(settings);
 	WINPR_ASSERT(src);
 
-	if (!settings->ServerMode)
-	{
-		settings->VirtualChannelCompressionFlags = src->VirtualChannelCompressionFlags;
+	/* MS servers and clients disregard in advertising what is relevant for their own side */
+	if (settings->ServerMode && settings->VirtualChannelCompressionFlags & VCCAPS_COMPR_SC &&
+	    src->VirtualChannelCompressionFlags & VCCAPS_COMPR_SC)
+		settings->VirtualChannelCompressionFlags |= VCCAPS_COMPR_SC;
+	else
+		settings->VirtualChannelCompressionFlags &= ~VCCAPS_COMPR_SC;
+
+	if (!settings->ServerMode && settings->VirtualChannelCompressionFlags & VCCAPS_COMPR_CS_8K &&
+	    src->VirtualChannelCompressionFlags & VCCAPS_COMPR_CS_8K)
+		settings->VirtualChannelCompressionFlags |= VCCAPS_COMPR_CS_8K;
+	else
+		settings->VirtualChannelCompressionFlags &= ~VCCAPS_COMPR_CS_8K;
+
+	/*
+	 * When one peer does not write the VCChunkSize, the VCChunkSize must not be
+	 * larger than CHANNEL_CHUNK_LENGTH (1600) bytes.
+	 * Also prevent an invalid 0 size.
+	 */
+	if (!settings->ServerMode &&
```

Comment:
some braces might help here

Reply:
I guess this about the individual conditions `src->VirtualChannelChunkSize > 16256` and `src->VirtualChannelChunkSize == 0`? In that case, did that in the last push(es).